hknweb/candidate/views.py

```diff
@@ -21,50 +48,87 @@ class CandRequestView(FormView, generic.ListView):
     context_object_name = 'challenge_list'
 
     def form_valid(self, form):
-        # This method is called when valid form data has been POSTed.
-        # It should return an HttpResponse.
         form.instance.requester = self.request.user
         form.save()
-        self.send_email(form)
+        self.send_request_email(form)
+        messages.success(self.request, 'Your request was submitted to the officer!')
         return super().form_valid(form)
 
-    def send_email(self, form):
+    def send_request_email(self, form):
         subject = 'Confirm Officer Challenge'
         officer_email = form.instance.officer.email
         text_content = 'Confirm officer challenge'
 
-        candidate_name = form.instance.requester.get_full_name()
-        candidate_username = form.instance.requester.username
-        # host = self.request.get_host()
-        link = self.request.build_absolute_uri(
+        confirm_link = self.request.build_absolute_uri(
                 reverse("candidate:challengeconfirm", kwargs={ 'pk' : form.instance.id }))
         html_content = render_to_string(
-            'candidate/email.html',
+            'candidate/request_email.html',
             {
-                'pk': form.instance.id,
-                'candidate_name' : candidate_name,
-                'candidate_username' : candidate_username,
-                'link' : link,
+                'candidate_name' : form.instance.requester.get_full_name(),
+                # TODO: for some usernames such as catherine.hu, this becomes a link. Why??
```

Comment:
catherine.hu becomes a link because it's got a dot in it. Not sure how to fix this. (Some kind of div wrapping in the HTML idk)

Reply:
I feel like there's nothing we can do, gmail just automatically detects it